azext_iot/common/base_discovery.py

```diff
@@ -178,9 +178,23 @@ def find_resource(self, resource_name: str, rg: str = None):
 
         resource_list = self.get_resources()
 
+        if isinstance(resource_list[0], dict):
```

Comment:
What is the reason for these changes, in the newer mgmt sdk the resource is a dict vs a model before?

Reply:
Good question - it may be a remain of when I tried to do modelless